src/controllers/materias.controller.js

```diff
@@ -0,0 +1,68 @@
+const Materias = require('../models/Materias')
+const Joi = require('joi')
+
+const getMateriaSchema = Joi.object({
+    id: Joi.string().guid().required(),
+    name: Joi.string().allow(''),
+  })
+  
+  const createMateriaSchema = Joi.object({
+    id: Joi.string().guid().required(),
+    name: Joi.string().required(),
+  })
+
+//get para obtener la materia
+const get_materia = async (req, res, next)=>{
+    try{
+        const {id, name} = req.body
+        
+        const { error } = getMateriaSchema.validate({ id, name })
+        if (error) return res.status(400).json({ error: error.details[0].message })
+        
+        if(id){
+            const findMateria = await Materias.findAll({
+                where:{id, name}
+            })
+            return res.json(findMateria)
+        }else{
+            return res.json({message: 'This materia do not exist'})
```

Comment:
Seria bueno mandar un numero de error, por ejemplo un 400
```suggestion
            return res.status(400).json({message: 'This materia do not exist'})
```